src/details/ArborX_DetailsNode.hpp

```diff
@@ -22,19 +22,27 @@ namespace Details
 {
 struct Node
 {
-  KOKKOS_INLINE_FUNCTION
-  Node() = default;
+  KOKKOS_FUNCTION
+  constexpr Node() = default;
 
-  KOKKOS_INLINE_FUNCTION
-  Node(const Kokkos::pair<Node *, Node *> &c, const Box &bb)
-      : children(c)
-      , bounding_box(bb)
+  KOKKOS_FUNCTION constexpr bool isLeaf() const noexcept
   {
+    return children.first == nullptr;
+  }
+
+  KOKKOS_FUNCTION std::size_t getLeafPermutationIndex() const noexcept
+  {
```

Comment:
Could do
```C++
assert(isLeaf());
```
here.  Result is garbage otherwise.

Reply:
Yeah, I had the same thought, and was not sure. I think it's fine as it is, as it does not change status quo.

Reply:
I decided to add it.